core/src/main/scala/org/broadinstitute/dsde/rawls/workspace/WorkspaceService.scala

```diff
@@ -624,7 +624,19 @@ class WorkspaceService(protected val ctx: RawlsRequestContext,
       } yield ()
     }
 
-  def assertNoChildrenBlockingWorkspaceDeletion(workspace: Workspace): Future[Unit] = for {
+  def assertNoGoogleChildrenBlockingWorkspaceDeletion(workspace: Workspace): Future[Unit] = for {
+    _ <-
+      if (workspace.googleProjectId.value.isEmpty) {
```

Comment:
If there's a cleaner way to write this if/else, definitely let me know. I spent some time Googling and didn't hit upon something better… though I suspect there is a way to avoid having to explicitly have successful "else" condition.

Reply:
It's more stylistic sugar, but `ApplicativeThrow` has many common exception-throwing patterns (though scala's type system makes this almost a ugly):
```scala 
ApplicativeThrow[Future].raiseWhen(condition: Bool)(t: Throwable)
```

Reply:
Thanks, I like it.